ratis-logservice/src/main/java/org/apache/ratis/logservice/api/LogName.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.ratis.logservice.api;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+/**
+ * Identifier to uniquely identify a {@link LogStream}.
+ */
+public class LogName {
+  // It's pretty likely that what uniquely defines a LogStream
+  // to change over time. We should account for this by making an
+  // API which can naturally evolve.
+  private final String name;
+
+  private LogName(String name) {
+    this.name = requireNonNull(name);
+  }
+
+  // Implementation detail -- we want uses to use the LogName as identifiable, not to
```

Comment:
s/Implementation detail/`@implSpec`?

Reply:
Nice! Didn't know about these. Happy to add that.

Reply:
Hrm, seems like these are proposed but not yet implemented? The maven-javadoc-plugin warns me against defining them.

```
Note: Custom tags that could override future standard tags:  @implSpec, @implNote. To avoid potential overrides, use at least one period character (.) in custom tag names.
```

Reply:
I see. This seems not enabled by default. I found a change in other projects to enable it via `maven-javadoc-plugin` see: https://github.com/CodeFX-org/demo-javadoc-8-tags/blob/master/pom.xml?ts=4#L110-L133

It's nice to have. No intention to block this PR. :)